diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f75adb..7182bca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
 
@@ -91,8 +90,7 @@ SOAPY_SDR_MODULE_UTIL(
 
 if(LibUSB_FOUND)
     # add usb streamer gadget
-    target_sources(PlutoSDRSupport PRIVATE PlutoSDR_RXStreamerUSBGadget.cpp)
-    target_sources(PlutoSDRSupport PRIVATE PlutoSDR_TXStreamerUSBGadget.cpp)
+   
 endif()
 
 ########################################################################
