diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f75adb..2e0ab76 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,7 +46,7 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" MATCH
     list(APPEND LibIIO_LIBRARIES -pthread)
 
     #disable warnings for unused parameters
-    add_compile_options(-Wall -Wextra -pedantic -Wno-unused-parameter)
+    add_compile_options(-Wall -Wextra -pedantic -Wno-unused-parameter -Wmismatched-new-delete)
 endif()
 
