diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f75adb..3fa0316 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,16 +37,16 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" MATCH
     include(CheckCXXCompilerFlag)
     CHECK_CXX_COMPILER_FLAG("-std=c++11" HAS_STD_CXX11)
     if(HAS_STD_CXX11)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra")
     else(HAS_STD_CXX11)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -Wc++11-extensions")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -Wc++11-extensions -Wall -Wextra")
     endif()
 
     #Thread support enabled (not the same as -lpthread)
     list(APPEND LibIIO_LIBRARIES -pthread)
 
     #disable warnings for unused parameters
-    add_compile_options(-Wall -Wextra -pedantic -Wno-unused-parameter)
+    add_compile_options(-Wall -Wextra -pedantic -Wno-unused-parameter -Wmismatched-new-delete)
 endif()
 
 
