diff --git a/support/test_c/CMakeLists.txt b/support/test_c/CMakeLists.txt
index 67e303b..da5256c 100644
--- a/support/test_c/CMakeLists.txt
+++ b/support/test_c/CMakeLists.txt
@@ -13,3 +13,5 @@ endif()
 add_executable(test_timestamp_loopback test_timestamp_loopback.c)
 
 target_link_libraries(test_timestamp_loopback ${SoapySDR_LIBRARIES})
+
+install(TARGETS test_timestamp_loopback DESTINATION ${RUNTIME_DIR} OPTIONAL)
